src/main/java/com/example/name/NameService.java

```diff
@@ -0,0 +1,28 @@
+package com.example.name;
+
+import org.springframework.stereotype.Service;
+import org.springframework.web.bind.annotation.RequestParam;
+
+import java.util.List;
+
+
+@Service
+public class NameService {
+
+    private final NameMapper nameMapper;
+
+    public NameService(NameMapper nameMapper) {
+        this.nameMapper = nameMapper;
+    }
+
+    public List<Name> getNames(@RequestParam(required = false) String startsWith) {
```

Comment:
`@RequestParam`はクエリ文字列をControllerに紐付けるのに必要なものなのでServiceの引数には不要ですね。
```suggestion
    public List<Name> getNames(String startsWith) {
```

Reply:
修正して、動作確認もできました。serviceを挟んだ場合、@RequestParamが必要なくなる理由がいまいちわからないです。必要になる場合はどんな処理の時ですか？

![image](https://github.com/daisuke-gk/task8/assets/146200380/ea4d0760-1c59-4924-b166-c17c3bc6392c)
![image](https://github.com/daisuke-gk/task8/assets/146200380/cfecf19f-4fe9-40c6-9eec-b31270cf21e3)

Reply:
`@RequestParam`というのは
`/dogs?type=small`
のようなクエリ文字列をControllerで扱うときに使うものだからですね。

Controllerでクエリ文字列を受け取り、それをServiceにわたすわけですから、Service側に `@RequestParam` をつける意味はないです。

Reply:
なるほど、ありがとうございます！